ignite/utils.py

```diff
@@ -125,3 +126,38 @@ def setup_logger(
         logger.addHandler(fh)
 
     return logger
+
+
+def one_rank_only(*args, **kwargs):
+    """Decorator to filter handlers wrt a rank number
+
+    .. code-block:: python
+        engine = ...
+
+        @engine.on(...)
+        @one_rank_only # equivalent @one_rank_only(rank=0)
+        def some_handler(_):
+            ...
+
+        @engine.on(...)
+        @one_rank_only(rank=1)
+        def some_handler(_):
+            ...
+    """
+    def _one_rank_only(fun):
```

Comment:
Maybe, we need to enclose the code with barriers:
https://pytorch.org/docs/stable/distributed.html#torch.distributed.barrier

Reply:
It means every procs should fire barrier event. Although I’m not sure that it is necessary. The code will be synchronised to next synchronous collective or wait. 

Btw, no comm is allowed in the handler.

Reply:
> Btw, no comm is allowed in the handler.

What do you mean by this ?

Reply:
Only one rank runs the code so no comm otherwise dead lock.